file-attribute-caching/src/test/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingFileSystemTests.kt

```diff
@@ -83,6 +84,21 @@ class AttributeCachingFileSystemTests {
         }
     }
 
+    @Test
+    fun `create java io tmpdir with jimfs wrapped by file attribute caching filesystem`() {
+        val filesystem = AttributeCachingFileSystem.wrapping(Jimfs.newFileSystem())
+        assertDoesNotThrow {
+            val javaTmpPath = filesystem.getPath(System.getProperty("java.io.tmpdir"))
+
+            if (!Files.exists(javaTmpPath)) {
+                Files.createDirectories(javaTmpPath)
+            }
+            assertThat(Files.exists(javaTmpPath)).isTrue()
+            Files.deleteIfExists(javaTmpPath)
+        }
+        filesystem.close()
```

Comment:
I've said this before, and this is a really great example of why `assertDoesNotThrow` is a bit silly.

If it throws, the test will fail with the exception, same as it would if your `assertDoesNotThrow` throws. 

The major difference with _not_ wrapping with `assertDoesNotThrow` is that it's cleaner, which is why I continuously advocate on behalf of it.

Reply:
it emphasizes the point of the test being the exception not being thrown, not the assertThat(Files.exists(javaTmpPath)).isTrue() line inside of it

Reply:
otherwise this test would have no asserts whatsoever and the outside reader would be left with a huh? why does this test even exist

Reply:
A similar effect could be achieved with the test name, but your point is recognized

Reply:
i will change the test names of this and the other test that uses doesNotThrow